### ISSUE TITLE
Terminating ... Transport endpoint is not connected

### ISSUE THREAD

**Issue by ScottSallinen** [EOSIO/eos#8296]
EOSIO v2.0.0rc2, net-threads set to 4, publicly reachable and frequently peered to from outside/unknown nodes. Behind the head block of the chain, and so it was syncing from peers.

Terminated while syncing up to head block from peers with the following message:

```
...
info  2019-12-07T01:55:08.306 net-3     net_plugin.cpp:1457           operator()           ] requesting range 93119840 to 93119939, from ****
terminating with uncaught exception of type boost::wrapexcept<boost::system::system_error>: shutdown: Transport endpoint is not connected
```

Stack trace as follows:

```
#0  __GI_raise (sig=sig@entry=6) at   ../sysdeps/unix/sysv/linux/raise.c:50
--
#1    0x00007fb88da2d535 in __GI_abort () at abort.c:79
#2    0x0000564d7d54a117 in abort_message ()
#3    0x0000564d7d54cf83 in demangling_terminate_handler() ()
#4    0x0000564d7d549a86 in std::__terminate(void (*)()) ()
#5    0x0000564d7d54bced in __cxa_throw ()
#6    0x0000564d7b502c55 in void   boost::throw_exception<boost::system::system_error>(boost::system::system_error   const&) ()
#7    0x0000564d7b797b15 in   boost::asio::basic_socket<boost::asio::ip::tcp,  boost::asio::executor>::shutdown(boost::asio::socket_base::shutdown_type)   ()
#8    0x0000564d7b7765ec in void  boost::asio::io_context::strand::dispatch<boost::asio::detail::binder1<boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},   boost::asio::io_context::strand>,boost::system::error_code>, std::__1::allocator<void>   >(boost::asio::detail::binder1<boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},   boost::asio::io_context::strand>,boost::system::error_code>&&,   std::__1::allocator<void> const&) const ()
#9    0x0000564d7b77724e in  boost::asio::detail::completion_handler<boost::asio::detail::binder1<boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},   boost::asio::io_context::strand>, boost::system::error_code>   >::do_complete(void*,   boost::asio::detail::scheduler_operation*, boost::system::error_code   const&, unsigned long) ()
#10 0x0000564d7b775fe0 in void  boost::asio::io_context::strand::dispatch<boost::asio::detail::binder1<boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},   boost::asio::io_context::strand>, boost::system::error_code>, std::__1::allocator<void>   >(boost::asio::detail::binder1<boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},   boost::asio::io_context::strand>, boost::system::error_code>&&,   std::__1::allocator<void> const&) const ()
#11 0x0000564d7b7758ac in boost::asio::detail::reactive_socket_accept_op<boost::asio::basic_socket<boost::asio::ip::tcp,  boost::asio::executor>, boost::asio::ip::tcp,  boost::asio::executor_binder<eosio::net_plugin_impl::start_listen_loop()::$_25::operator()()   const::{lambda(boost::system::error_code)#1},boost::asio::io_context::strand>,   boost::asio::detail::io_object_executor<boost::asio::executor>  >::do_complete(void*, boost::asio::detail::scheduler_operation*, boost::system::error_code   const&, unsigned long) ()
#12 0x0000564d7b5061ba in boost::asio::detail::scheduler::do_run_one(boost::asio::detail::conditionally_enabled_mutex::scoped_lock&,   boost::asio::detail::scheduler_thread_info&,   boost::system::error_code const&) ()
#13 0x0000564d7b505c61 in   boost::asio::detail::scheduler::run(boost::system::error_code&) ()
#14 0x0000564d7c2dac5b in boost::asio::detail::executor_op<boost::asio::detail::work_dispatcher<eosio::chain::named_thread_pool::named_thread_pool(std::__1::basic_string<char,   std::__1::char_traits<char>, std::__1::allocator<char> >,   unsigned long)::$_0>, std::__1::allocator<void>,   boost::asio::detail::scheduler_operation>::do_complete(void*,   boost::asio::detail::scheduler_operation*, boost::system::error_code   const&, unsigned long) ()
#15 0x0000564d7b5061ba in   boost::asio::detail::scheduler::do_run_one(boost::asio::detail::conditionally_enabled_mutex::scoped_lock&,   boost::asio::detail::scheduler_thread_info&,   boost::system::error_code const&) ()
#16 0x0000564d7b505c61 in   boost::asio::detail::scheduler::run(boost::system::error_code&) ()
#17 0x0000564d7c2db4f6 in   boost::asio::detail::posix_thread::func<boost::asio::thread_pool::thread_function>::run()   ()
#18 0x0000564d7b505a3d in   boost_asio_detail_posix_thread_function ()
#19 0x00007fb88e19a182 in start_thread   (arg=<optimized out>) at pthread_create.c:486
#20 0x00007fb88db25b1f in clone () at   ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by heifner** [comment]
This is very helpful. Thanks for the report.

**Comment by heifner** [comment]
Resolved by #8303
